### PR TITLE
Rewrite RenderPipelineDescriptor according to aspects

### LINKED ISSUE #1166: Update RenderPipelineDescriptor API according to aspects
The upstream API was recently changed in https://github.com/gpuweb/gpuweb/pull/1352
Let's update our side, just in time for 0.7 release.

### PR DESCRIPTION
**Pull request by kvark**
**Connections**
Fixes #1166

**Description**
Totally re-imagines the structure tree for describing the rendering pipeline, based on the aspects instead of logical steps.

**Testing**
Just local testing. I don't think there is any concern that these changes might not work, they just need the clients to adapt.

### CODE REVIEW

File: wgpu-types/src/lib.rs [hunk 1]
```diff
@@ -2510,6 +2658,7 @@ pub enum BindingType {
 }
 
 impl BindingType {
+    /// Returns true for buffer bindings with dynamic offseting enabled.
```
Comment (grovesNL) [inline wgpu-types/src/lib.rs:2661]: nit: offsetting or offset

**Review by grovesNL — APPROVED** [review]
Looks great! Thank you for updating this

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,26 @@
 # Change Log
 
+## v0.7 (TBD)
+  - Major API changes:
+    - `RenderPipelineDescriptor`
+    - `BindingType`
+  - Features:
+    - (beta) WGSL support, including the ability to bypass SPIR-V entirely
+    - (beta) implicit bind group layout support
+    - timestamp queries and pipeline statistics
```
Comment (cwfitzgerald) [inline CHANGELOG.md:10]: nit:

```suggestion
    - timestamp and pipeline statistics queries
```

**Review by cwfitzgerald — APPROVED** [review]
LGTM! One minor nit

**Comment by kvark** [comment]
Thank you for the reviews!
bors r=grovesNL,cwfitzgerald

**Pull request merged**